app/lib/payment_gateways/stripe_crypt.rb

```diff
@@ -36,21 +32,41 @@ def customer
       @customer ||= find_or_create_customer
     end
 
+    def update_billing_address(billing_address)
+      begin
+        Stripe.api_key = api_key  # Set actual Stripe secret key
+        latest_payment_method_id = latest_payment_method_id_for_customer
+        return true unless latest_payment_method_id.present?
+        payment_method = Stripe::PaymentMethod.retrieve(payment_method_id)
+        update_and_save_billing_details(payment_method, billing_address)
+      rescue Stripe::StripeError => stripe_error
+        handle_stripe_error(stripe_error)
+      ensure
+        reset_stripe_api_key
```

Comment:
@nidhi-soni1104 Can you explain, why do we need to set and unset the Stripe key this way?... Can't we just keep passing it to `#retrieve` as a second argument?

Reply:
I did this because when I tried updating on stripe passing Stripe key  as second argument, it didn't get updated and then I tried it n number of times with other different ways but only this method worked for me .